src/libp2p_framed_stream.erl

```diff
@@ -352,14 +473,35 @@ handle_fdset(State=#state{connection=Connection}) ->
 handle_resp_send(Action, Data, State=#state{}) ->
     handle_resp_send(Action, Data, ?SEND_TIMEOUT, State).
 
+-spec send_key(binary(), #state{}) -> #state{}.
+send_key(Data, #state{send_pid=SendPid}=State) ->
+    Key = make_ref(),
+    Bin = <<(byte_size(Data)):32/little-unsigned-integer, Data/binary>>,
+    SendPid ! {send, Key, Bin},
+    State.
+
+
 -spec handle_resp_send(send_result_action(), binary(), non_neg_integer(), #state{}) -> #state{}.
-handle_resp_send(Action, Data, Timeout, State=#state{sends=Sends, send_pid=SendPid}) ->
+handle_resp_send(Action, Data, Timeout, State=#state{secured=true, exchanged=true,
+                                                    send_nonce=Nonce}) ->
+    case enacl:aead_chacha20poly1305_encrypt(State#state.send_key, Nonce, <<>>, Data) of
+        {error, _Reason} ->
+            lager:warning("failed to encrypt ~p : ~p", [{Nonce, Data}, _Reason]),
+            State;
```

Comment:
Should we close the connection if this happens?

Reply:
maybe?

Reply:
yup, good catch